usaspending_api/common/api_request_utils.py

```diff
@@ -219,6 +219,33 @@ def range_intersect(self, fields, values):
         return Q(**q_case)
 
 
+# Handles unique value requests
+class UniqueValueHandler():
+    @staticmethod
+    # Data set to use (should be filtered already)
+    # Fields to find unique values for
+    # Returns a dictionary containing fields, values and their counts
+    # {
+    #   "recipient__name": {
+    #       "Jon": 5,
+    #       "Joe": 2
+    #   }
+    # }
+    def get_values_and_counts(data_set, fields):
+        data_set = data_set.all()  # Do this because we don't want to get finnicky with annotations
+        response_object = {}
+        if fields:
+            for field in fields:
```

Comment:
Maybe this is what you're alluding too in the annotations comment above...forgive the question since I'm new to the codebase and am still refreshing my good old Django ORM skills, but does it make sense to issue a count unique query for reach field in the incoming fields list?

Again, that's an honest question as I get up to speed on the project!